frontend/public/components/dashboards-page/overview-dashboard/health-card.tsx

```diff
@@ -0,0 +1,176 @@
+  } else if (subsystemBySeverity.warning.length > 0) {
+    healthState =
+      subsystemBySeverity.warning.length === 1
+        ? subsystemBySeverity.warning[0]
+        : { state: HealthState.WARNING, message: 'Multiple warnings', details: 'Cluster health is degraded' };
+  }
+
+  return healthState;
+};
+
+const getName = (isOpenShift: boolean): string => isOpenShift ? getBrandingDetails().productName : 'Kubernetes';
+
+const getK8sHealthState = (isOpenShift: boolean, k8sHealth: any): SubsystemHealth => {
+  if (!k8sHealth) {
+    return { state: HealthState.LOADING };
+  }
+  return k8sHealth === 'ok'
+    ? { message: `${getName(isOpenShift)} ${HEALTHY}`, state: HealthState.OK }
+    : { message: `${getName(isOpenShift)} ${ERROR}`, state: HealthState.ERROR };
+};
+
+const mapStateToProps = (state: RootState) => ({
+  isOpenShift: state[featureReducerName].get(FLAGS.OPENSHIFT),
+});
+
+const fetchK8sHealth = async(url) => {
+  const response = await coFetch(url);
+  return response.text();
+};
+
+const _HealthCard: React.FC<HealthProps> = ({
+  watchURL,
+  stopWatchURL,
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  urlResults,
+  prometheusResults,
+  isOpenShift,
+}) => {
+  React.useEffect(() => {
+    const subsystems = plugins.registry.getDashboardsOverviewHealthSubsystems();
+    watchURL('healthz', fetchK8sHealth);
+
+    subsystems.filter(plugins.isDashboardsOverviewHealthURLSubsystem).forEach(subsystem => {
+      const { url, fetch } = subsystem.properties;
+      watchURL(url, fetch);
+    });
+    subsystems.filter(plugins.isDashboardsOverviewHealthPrometheusSubsystem).forEach(subsystem => {
+      const { query } = subsystem.properties;
+      watchPrometheus(query);
+    });
+    return () => {
+      stopWatchURL('healtz');
+
+      subsystems.filter(plugins.isDashboardsOverviewHealthURLSubsystem).forEach(subsystem =>
+        stopWatchURL(subsystem.properties.url)
+      );
+      subsystems.filter(plugins.isDashboardsOverviewHealthPrometheusSubsystem).forEach(subsystem =>
+        stopWatchPrometheusQuery(subsystem.properties.query)
+      );
+    };
+  }, [watchURL, stopWatchURL, watchPrometheus, stopWatchPrometheusQuery]);
+
+  const subsystems = plugins.registry.getDashboardsOverviewHealthSubsystems();
+  const k8sHealth = urlResults.getIn(['healthz', 'result']);
+  const k8sHealthState = getK8sHealthState(isOpenShift, k8sHealth);
+
+  const subsystemsHealths = subsystems.map(subsystem => {
+    let result;
+    if (plugins.isDashboardsOverviewHealthPrometheusSubsystem(subsystem)) {
+      result = prometheusResults.getIn([subsystem.properties.query, 'result']);
+    } else {
+      result = urlResults.getIn([subsystem.properties.url, 'result']);
+    }
+    return subsystem.properties.healthHandler(result);
+  });
+
+  const healthState = getClusterHealth([k8sHealthState, ...subsystemsHealths]);
+
+  let seeAll;
```

Comment:
I think it's cleaner to make this an expression directly in the markup from the `return` value below.

Reply:
Sounds good to me. Its now part of `return` statement